plugin/src/test/java/io/jenkins/plugins/analysis/warnings/steps/StepsITest.java

```diff
@@ -191,6 +191,38 @@ void shouldSkipBlaming() {
         assertThat(getConsoleLog(baseline)).contains("Skipping SCM blames as requested");
     }
 
+    /**
+     * Runs a pipeline and verifies that logging is not suppressed.
+     */
+    @Test
+    void shouldLog() {
+        WorkflowJob job = createPipelineWithWorkspaceFilesWithSuffix("checkstyle1.xml");
+
+        job.setDefinition(new CpsFlowDefinition("node {\n"
+                + "  stage ('Integration Test') {\n"
+                + "         recordIssues quiet: false, tool: checkStyle(pattern: '**/" + "checkstyle1" + "*')\n"
+                + "  }\n"
+                + "}", true));
+        Run<?, ?> baseline = buildSuccessfully(job);
+        assertThat(getConsoleLog(baseline)).contains("Searching for all files in");
+    }
+
+    /**
+     * Runs a pipeline and verifies that logging is suppressed.
+     */
+    @Test
+    void shouldSuppressLog() {
+        WorkflowJob job = createPipelineWithWorkspaceFilesWithSuffix("checkstyle1.xml");
+
+        job.setDefinition(new CpsFlowDefinition("node {\n"
+                + "  stage ('Integration Test') {\n"
+                + "         recordIssues quiet: true, tool: checkStyle(pattern: '**/" + "checkstyle1" + "*')\n"
+                + "  }\n"
+                + "}", true));
+        Run<?, ?> baseline = buildSuccessfully(job);
+        assertThat(getConsoleLog(baseline)).hasSize(0);
```

Comment:
I think there is no way to suppress or work on the whole log. I think you simply should 
```
assertThat(getConsoleLog(baseline)).doesNotContain("Searching for all files in");
```

You can also use a `ParameterizedTest` with a `ValueSource(booleans = {true, false}` and then use the same test: 
```
if (isQuite) {
   assertThat(getConsoleLog(baseline)).doesNotContain("Searching for all files in");
} 
else {
   assertThat(getConsoleLog(baseline)).contains("Searching for all files in");
}
```

Reply:
Not sure I get it right, does console log really contains `LogHandler` output or its feeded separately without `LogHandler`? Whatever I've tried console contained all content.
Also, the other option add & check new log line would make it not completely muted (not a big issue, but...)

Reply:
Should be fixed now, there was still one method left that had no guard. Now the guard is at the lowest method in the call stack.